File: pset6/vigenere/vigenere.py
from cs50 import get_string
import sys


"""Checking for proper usage"""

if len(sys.argv) != 2:
    sys.exit("Usage: python vigenere.py k")

if not sys.argv[1].isalpha():
    sys.exit("Usage: python vigenere.py k")


Key = sys.argv[1].lower()


"""Building an alphabet list and a letter-->number dictionary for later"""

alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l",
            "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"]

alpha_dict = {letter: idx for idx, letter in enumerate(alphabet)}

MSG = get_string("plaintext: ")
ITL = len(Key)
MSG2 = ""
space = 0

for i in range(len(MSG)):
    if MSG[i] == " ":
        space += 1
    if MSG[i].isalpha():
        if MSG[i].isupper():
            MSG2 += alphabet[(alpha_dict[MSG[i].lower()] + alpha_dict[Key[(i + space) % ITL]]) % 26].upper()
        else:
            MSG2 += alphabet[(alpha_dict[MSG[i]] + alpha_dict[Key[(i + space) % ITL]]) % 26]

    else:
        MSG2 += MSG[i]

print("ciphertext:", MSG2)

File: pset2/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int shift(char c);

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./vigenere keyword\n");
        return 1;
    }
    string codeword = argv[1];
    int i = 0;
    while (i < strlen(codeword))
        {
        if ((codeword[i] < 'A') || (codeword[i] > 'Z' && codeword[i] < 'a') || (codeword[i] > 'z'))
            {
            printf("Usage: ./vigenere keyword\n");
            return 1;
            }
        i = i + 1;
        }
    string message = get_string("plaintext: ");
    printf("ciphertext: ");
    int b = 0;
    int cshift = 0;
    while (b < strlen(message))
    {
        int key = shift(codeword[cshift%strlen(codeword)]);
        if (message[b] >= 'a' && message[b] <= 'z')
            {
            printf("%c", (((message[b] - 'a') + key) % 26) + 'a');
            cshift = cshift + 1;
            }
        else if (message[b] >= 'A' && message[b] <= 'Z')
            {
            printf("%c", (((message[b] - 'A') + key) % 26) + 'A');
            cshift = cshift + 1;
            }
        else
            {
            printf("%c", message[b]);
            }
        b = b + 1;
    }
    printf("\n");
    return 0;
}

int shift(char c)
{
    int shift = 0;
   if (c >= 'a' && c <= 'z')
       {
       shift = c - 'a';
       }
   if (c >= 'A' && c <= 'Z')
       {
       shift = c - 'A';
       }
   return shift;
}
